src/coprocessor/endpoint.rs

```diff
@@ -59,24 +61,46 @@ const OUTDATED_ERROR_MSG: &'static str = "request outdated.";
 
 const ENDPOINT_IS_BUSY: &'static str = "endpoint is busy";
 
-pub struct Host {
+pub struct Host<R: RaftStoreRouter + 'static> {
     engine: Box<Engine>,
     sched: Scheduler<Task>,
     reqs: HashMap<u64, Vec<RequestTask>>,
     last_req_id: u64,
-    pool: ThreadPool<CopContext>,
-    low_priority_pool: ThreadPool<CopContext>,
-    high_priority_pool: ThreadPool<CopContext>,
+    pool: ThreadPool<CopContext<R>>,
+    low_priority_pool: ThreadPool<CopContext<R>>,
+    high_priority_pool: ThreadPool<CopContext<R>>,
     max_running_task_count: usize,
 }
 
+pub type CopRequestStatistics = HashMap<u64, FlowStatistics>;
+
+struct CopContextFactory<R: RaftStoreRouter + 'static> {
+    raft_router: R,
+}
+
+impl<R> ContextFactory<CopContext<R>> for CopContextFactory<R>
+where
+    R: RaftStoreRouter + 'static,
```

Comment:
IMO, we should not introduce RaftStoreRouter to the coprocessor, seem strange here. 

You can define a trait which sending your read stats like storage raftkv does.